lib/chart_helpers.dart

```diff
@@ -93,3 +127,77 @@ ChartConfiguration generateBarChartConfig(
       data: dataset,
       options: _generateBarChartOptions(dataNormalisationEnabled));
 }
+
+ChartOptions _generateTimeSeriesChartOptions(bool dataNormalisationEnabled) {
+  var labelPrepend = dataNormalisationEnabled ? 'Percentage' : 'Number';
+  var labelString = '${labelPrepend} of interactions';
+  var chartX = ChartXAxe()
+    ..stacked = false
+    ..ticks = (LinearTickOptions()
+      ..minRotation = 0
+      ..maxRotation = 90);
+
+  var chartY = ChartYAxe()
+    ..stacked = false
+    ..scaleLabel = ScaleTitleOptions(labelString: labelString, display: true)
+    ..ticks = (LinearTickOptions()..min = 0);
+
+  var tooltipLabelCallback = (ChartTooltipItem tooltipItem, ChartData data) {
+    var xLabel = data.datasets[tooltipItem.datasetIndex].label;
+    var yLabel = tooltipItem.yLabel;
+    var suffix = dataNormalisationEnabled ? '%' : '';
+    return '${xLabel}: ${yLabel}${suffix}';
+  };
+
+  return ChartOptions(
+      responsive: true,
+      tooltips: ChartTooltipOptions(
+          mode: 'index',
+          callbacks:
+              ChartTooltipCallback(label: allowInterop(tooltipLabelCallback))),
+      legend: ChartLegendOptions(
+          position: 'bottom', labels: ChartLegendLabelOptions(boxWidth: 12)),
+      scales: ChartScales(display: true, xAxes: [chartX], yAxes: [chartY]));
+}
+
+ChartDataSets _generateTimeSeriesChartDataset(
+    String label, List<num> data, String lineColor) {
+  return ChartDataSets(
+      label: label,
+      fill: false,
+      backgroundColor: lineColor + 'aa',
+      borderColor: lineColor,
+      hoverBackgroundColor: lineColor,
+      hoverBorderColor: lineColor,
+      borderWidth: 1,
+      lineTension: 0,
+      data: data);
+}
+
+ChartConfiguration generateTimeSeriesChartConfig(
+    model.Chart chart, bool dataNormalisationEnabled) {
+  var colors = chart.colors ?? lineChartDefaultColors;
+  var datasets = chart.fields.asMap().entries.map((entry) {
+    var index = entry.key;
+    var field = entry.value;
+    return _generateTimeSeriesChartDataset(field.label ?? field.field,
+        field.time_bucket.values.toList(), colors[index]);
+  }).toList();
+
+  return ChartConfiguration(
+      type: 'line',
+      data: ChartData(
+          labels: chart.fields.first.time_bucket.keys.map((key) {
+            var date = DateTime.parse(key);
+            switch (chart.timestamp.aggregate) {
+              case model.TimeAggregate.day:
+                return DateFormat('dd MMM').format(date);
+              case model.TimeAggregate.hour:
+                return DateFormat('dd MMM h:mm a').format(date);
+              default:
+                return DateFormat('dd MMM h:mm a').format(date);
```

Comment:
nit: you could probably merge these two cases together:
```suggestion
              case model.TimeAggregate.hour:
              default:
                return DateFormat('dd MMM h:mm a').format(date);
```

Reply:
Fixed